src/Microsoft.Extensions.Logging.Abstractions/Internal/FormattedLogValues.cs

```diff
@@ -19,12 +19,23 @@ public class FormattedLogValues : IReadOnlyList<KeyValuePair<string, object>>
         private readonly LogValuesFormatter _formatter;
         private readonly object[] _values;
         private readonly string _originalMessage;
+        private const int MaxCachedFormatters = 1024;
 
         public FormattedLogValues(string format, params object[] values)
         {
             if (values?.Length != 0 && format != null)
             {
-                _formatter = _formatters.GetOrAdd(format, f => new LogValuesFormatter(f));
+                if (_formatters.Count >= MaxCachedFormatters)
```

Comment:
So if we are over the limit we just create it? That seems bad, what's the cache eviction policy on this? Seems like frequently occurring formats should never be dropped.